### PR TITLE
created explosion when bat dies

### PR DESCRIPTION
**Pull request by ishakanodia**
Explosion effect when bat dies- but the explosion is huge - need to change that png

### CODE REVIEW

File: Scenes/Monsters/Bat.gd [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
+	knockback = knockback.move_toward(Vector2.ZERO, 200 * delta)
+	knockback = move_and_slide(knockback)
 
 
 func _on_Hurtbox_area_entered(area):
-	#knockback = area.knockback_vector * 120
-	queue_free()
+	stats.health -= 1
+	
+	knockback = Vector2.RIGHT * 120
```
Comment (josejulio) [inline Scenes/Monsters/Bat.gd:22]: I guess here we need to compute the direction vector. If my math is not that rusty, a (area.global_position - global_position).normalize() would do the trick

**Review by justinorringer — APPROVED** [review]
This looks good! 

Once its merged in, I'll integrate the Bat script with Enemy so it follows the player too.

And the tutorial has a solution for the knockback direction, but even he doesn't like it. https://www.youtube.com/watch?v=mm4D1ETclAc&ab_channel=HeartBeast

File: Scripts/Player/Entity.gd [hunk 2]
```diff
@@ -11,28 +11,33 @@ enum {
...
 		 animation_state = animation_tree.get("parameters/playback")
-
+	
+	
 func _physics_process(delta):
 	pass
 
 func move_state(delta, movement_vector):
-	
+	var input_vector = Vector2.ZERO
```
Comment (josejulio) [inline Scripts/Player/Entity.gd:31]: I guess that this is not needed in this PR?

File: Scenes/Monsters/Bat.gd (hunk 1 above)
Comment (josejulio) [reply Scenes/Monsters/Bat.gd:22]: Might need to update the order depending if this is applying a knockback to the player or the bat :D

File: Scripts/Player/Entity.gd (hunk 2 above)
Comment (justinorringer) [reply Scripts/Player/Entity.gd:31]: Yeah, that's outdated. Wonder how it got back in there haha.

Comment (ishakanodia) [reply Scripts/Player/Entity.gd:31]: Yeah sorry I was trying something there, forgot to remove it

**Pull request merged**